### DISCUSSION TITLE
http-restful-server example compiles, opens listeners, but results in tis not enabled error messages

### DISCUSSION THREAD

**Discussion by schilton** [cesanta/mongoose#2181] [Ask questions, get answers] [answered]
I have tried to run the http-restful-server example and I can get it to compile and it runs as expected on port 80.  It also says that it is listening on both port 80 and port 443.  However, when you try to access over SSL, I receive the error message "TLS is not enabled."  Based on my understanding of the tls enablement procedures, my attached Makefile enables SSL.  Does anyone have any suggestions on what I should try next?
[Makefile.txt](https://github.com/cesanta/mongoose/files/11409512/Makefile.txt)
[main.txt](https://github.com/cesanta/mongoose/files/11409513/main.txt)

**Comment by scaprile** [comment, marked as answer]
> my attached Makefile enables SSL.

I beg to differ

> Does anyone have any suggestions on what I should try next?

Please read the docs and tutorials, lines 11 and 12 on your very same main.txt file
https://mongoose.ws/documentation/tutorials/tls/#how-to-build

**Reply by schilton** [reply]
Although I am unsure why it worked with just cc and not my make file, thank you.  While the examples provides for both secure and unsecure connections, how would you force an SSL connection?  There used to be a url_rewrite function, but I do not see that anymore.
